e2e/tests/stake_app_transfer.feature

```diff
@@ -2,21 +2,37 @@ Feature: App Stake Transfer Namespace
         Given the user has the pocketd binary installed
-	# Unstake the applications in case they're already staked.
-        And this test ensures the "application" for account "app2" is not staked
-        And this test ensures the "application" for account "app3" is not staked
+        And an account exists for "app3"
         # Stake with 1 uPOKT more than the current stake used in genesis to make
         # the transaction succeed.
         And the account "app2" has a balance greater than "1000070" uPOKT
-        And an account exists for "app3"
         And the user successfully stakes a "application" with "1000070" uPOKT for "anvil" service from the account "app2"
+        # Begin transfer
         When the user transfers the "application" stake from account "app2" to account "app3"
         Then the user should be able to see standard output containing "txhash:"
         And the user should be able to see standard output containing "code: 0"
         And the pocketd binary should exit without error
         And the user should wait for the "application" module "TransferApplication" message to be submitted
+        And the user should wait for the "application" module "TransferBegin" tx event to be broadcast
+        # The source application should still be staked and in the transfer period
+        And the "application" for account "app2" is staked with "1000070" uPOKT
+        And the application for account "app2" is in the "transfer" period
+        # The destination application is not created until the transfer period ends
+        And the user verifies the "application" for account "app3" is not staked
+        And the user should wait for the "application" module "TransferEnd" end block event to be broadcast
         And the "application" for account "app3" is staked with "1000070" uPOKT
-        # TODO_IN_THIS_PR: fix idempotency issue with this scenario (this step passes on the first run but fails on subsequent runs)
         And the account balance of "app3" should be "0" uPOKT "less" than before
+        # The source application should be unstaked with no account balance
+        # change after the transfer period
         And the user verifies the "application" for account "app2" is not staked
         And the account balance of "app2" should be "0" uPOKT "more" than before
+        # Cleanup for other tests
+        # TODO(#): Until a network state reset API is implemented, we SHOULD
+        # manually unstake the applications to mitigate failures in other features
+        # which may otherwise make false assumptions about the starting state.
+        When the user successfully unstakes a "application" from the account "app3"
+
+#    TODO_TEST: Scenario: User cannot start an Application stake transfer from Application which has a pending transfer
+#    TODO_TEST: Scenario: Application stake transfer fails if the destination Application stakes before the transfer period elapses
```

Comment:
TODO_TEST: If two applications start a transfer to the same destination, one of them fails and the other succeeds?

Would love to have this test done in this PR.

Reply:
:ok_hand: :100: